plugins/network-elements/nuage-vsp/src/com/cloud/network/element/NuageVspElement.java

```diff
@@ -194,7 +224,23 @@ public boolean configure(String name, Map<String, Object> params) throws Configu
     @Override
     public boolean implement(Network network, NetworkOffering offering, DeployDestination dest, ReservationContext context) throws ConcurrentOperationException,
             ResourceUnavailableException, InsufficientCapacityException {
-        s_logger.debug("Entering NuageElement implement function for network " + network.getDisplayText() + " (state " + network.getState() + ")");
+        if (s_logger.isDebugEnabled()) {
+            s_logger.debug("Entering NuageElement implement function for network " + network.getDisplayText() + " (state " + network.getState() + ")");
+        }
+
+        if (network.getVpcId() != null) {
+            List<NetworkACLItemVO> rules;
+            if (network.getNetworkACLId() != null) {
+                rules = _networkACLItemDao.listByACL(network.getNetworkACLId());
+            } else {
+                rules = Collections.emptyList();
+            }
```

Comment:
Changed parameter of <code>NetworKACLItemDAO.listByACL(long)</code> to <code>Long</code>, and returning an empty list in case of <code>null</code>.